aat/main_test.go

```diff
@@ -157,7 +170,41 @@ func TestMain(m *testing.M) {
 	case "":
 		serType = ""
 		sockDesc = "LOCAL CONNECTIONS"
-	case "http", "ws":
+	case "http", "https":
+		s := router.NewWebsocketServer(nxr)
+		sockDesc = "WEBSOCKETS"
+		// Set optional websocket config.
+		if compress {
+			s.Upgrader.EnableCompression = true
+			sockDesc += " + compression"
+		}
+		s.EnableTrackingCookie = true
+		s.EnableRequestCapture = true
+		s.KeepAlive = time.Second
+
+		// Run the HTTP/HTTPS server in a go routine since ListenAndServe blocks
+		go func() {
+			// Create a router and assign the serveHTTP handler to /ws endpoint
+			webRouter := mux.NewRouter()
+			webRouter.HandleFunc("/ws", s.ServeHTTP).Methods("GET")
+			wampServer := &http.Server{
+				Addr:    httpAddr,
+				Handler: webRouter,
+			}
+
+			// Check if we should use the certs when using tls
+			if scheme == "http" {
+				err = wampServer.ListenAndServe()
```

Comment:
Won't this cause a race condition with different goroutines writing to the same `err`?  Maybe assign to something other than `err`  in goroutine.